backend/models/planInvite.ts

```diff
@@ -45,16 +45,30 @@ export default class PlanInviteModel {
     }
   }
 
-  public static async readOnePlanInvite(planID: string, email: string): Promise<PlanInvite | null> {
+  public static async readOnePlanInvite(params: PlanInviteModelReadInput, options?: 'AND' | 'OR'): Promise<PlanInvite | null> {
     try {
+      if (options === 'OR') {
+        const responseData = await db.planInvite.findFirst({
+          where: {
+            OR: {
+              ...params
+            }
+          }
+        })
+
+        if (responseData === null) throw new Error(`Unable to find a matching plan invite with in the database${JSON.stringify(params)}`)
+
+        return responseData;
+      }
```

Comment:
The simplification you can make is to take advantage of having "AND" be the default.

so you could remove all of this by simply stating

```
where: {
   [options]: {
      ...params
    }
}
```
and then default your `options` to `AND`
```  public static async readOnePlanInvite(params: PlanInviteModelReadInput, options?: 'AND' | 'OR' = 'AND'): Promise<PlanInvite | null> {```

or whatever the syntax is i always forget w/ typescript